ArcadiaParties.API/CustomMiddlewares/DatabaseAuthorizationMiddleware.cs

```diff
@@ -30,7 +29,7 @@ public async Task InvokeAsync(HttpContext context, IMediator mediator)
             }
 
             var newIdentity = new ClaimsIdentity(
-                NegotiateDefaults.AuthenticationScheme,
+                user.Identity.AuthenticationType,
```

Comment:
I think it would be useful to check whether user is authenticated, before checking their roles. And if not - middleware should set response status to 401.

Reply:
Check has been added